vaultlocker/shell.py

```diff
@@ -40,10 +40,17 @@ def _vault_client(config):
     """
     client = hvac.Client(
         url=config.get('vault', 'url'),
-        verify=config.get('vault', 'ca_bundle', fallback=True)
+        token=config.get('vault', 'token', fallback=None),
+        verify=config.get('vault', 'ca_bundle', fallback=True),
+        namespace=config.get('vault', 'namespace', fallback=None)
     )
-    client.auth_approle(config.get('vault', 'approle'),
-                        secret_id=config.get('vault', 'secret_id'))
+    if not config.get('vault', 'role_id', fallback=None) is None:
+        role_id=config.get('vault', 'role_id')
+    if not config.get('vault', 'approle', fallback=None) is None:
+        role_id=config.get('vault', 'approle')
+    client.auth.approle.login(role_id,
+                              secret_id=config.get('vault', 'secret_id', fallback=None),
+                              mount_point=config.get('vault', 'mount_point', fallback="approle"))
```

Comment:
I'm note sure I understand the use of 'approle' in the fallback context here?

Reply:
Because it was the default value before, so as to keep compatibility with previous versions

Reply:
approle was deprecated in preference to role_id - this is the mount_point configuration

Reply:
You're right, I'll remove that fallback